thirdparty/github.go

```diff
@@ -476,34 +478,35 @@ func SendPendingStatusToGithub(input SendGithubStatusInput) error {
 // GitHub repo and returns the merge base commit's SHA.
 func GetGithubMergeBaseRevision(ctx context.Context, token, owner, repo, baseRevision, currentCommitHash string) (string, error) {
 	installationToken, _ := GetInstallationToken(ctx, owner, repo, nil)
+	if installationToken != "" {
+		httpClient := getGithubClientRetry(installationToken, "GetGithubMergeBaseRevision")
+		defer utility.PutHTTPClient(httpClient)
+		client := github.NewClient(httpClient)
 
-	httpClient := getGithubClientRetry(installationToken, "GetGithubMergeBaseRevision")
-	defer utility.PutHTTPClient(httpClient)
-	client := github.NewClient(httpClient)
-
-	compare, resp, err := client.Repositories.CompareCommits(ctx, owner, repo, baseRevision, currentCommitHash, nil)
-	if resp != nil {
-		defer resp.Body.Close()
-		if err == nil && compare != nil && compare.GetMergeBaseCommit() != nil && compare.GetMergeBaseCommit().GetSHA() != "" {
-			return compare.GetMergeBaseCommit().GetSHA(), nil
+		compare, resp, err := client.Repositories.CompareCommits(ctx, owner, repo, baseRevision, currentCommitHash, nil)
+		if resp != nil {
+			defer resp.Body.Close()
+			if err == nil && compare != nil && compare.GetMergeBaseCommit() != nil && compare.GetMergeBaseCommit().GetSHA() != "" {
```

Comment:
same thought here

Reply:
it seems a little weird to check the comp before checking nil but there shouldn't be a problem compare is always nil if there's an error and this is very temporary